### ISSUE TITLE
Test for and fix encoding errors on Windows

### ISSUE THREAD

**Issue by lptr** [gradle/gradle#25322]
This is a followup to https://github.com/gradle/gradle/pull/25261.

Let's enable the use of non-ASCII paths for Windows integration tests, and fix any discovered problems.

**Comment by lptr** [comment]
Actually, those code points stand for something completely different... 🤔

But I tried running the child process from Java with `ProcessBuilder`, using the explicit character code `\u0219` and got:

```text
> Sending:  te?t, chars: [ u0074 u0065 u0219 u0074 ], bytes: [ 74 65 c8 99 74 ]
> Received: te?t, chars: [ u0074 u0065 u003f u0074 ], bytes: [ 74 65 3f 74 ]
```

It doesn't matter if I use `cmd`:

```java
new ProcessBuilder("cmd", "/u", "/c", "java", "-cp", ".", "-Dvar=" + value, "Main")
```

(with or without `/u`)

Or execute Java directly:

```java
new ProcessBuilder("java", "-cp", ".", "-Dvar=" + value, "Main")
```

One option I can imagine working is to encode the parameters ourselves with some ASCII-only encoding. URI encoding comes to mind, but `%` would need to be double-escaped...

**Comment by lptr** [comment]
Also note that if I try to send non-ASCII characters that _are_ part of CP1252, like `tést`, the transmission works (though the console output is still garbled, the codepoints are correct):

```text
> Sending:  t??st, chars: [ u0074 u00c3 u00a9 u0073 u0074 ], bytes: [ 74 c3 83 c2 a9 73 74 ]
> Received: t??st, chars: [ u0074 u00c3 u00a9 u0073 u0074 ], bytes: [ 74 c3 83 c2 a9 73 74 ]
```

**Comment by lptr** [comment]
Using `chcp 65001` to set the codepage to something unicode-i fixes my console, but doesn't fix the process invocation:

```text
> Sending:  teșt, chars: [ u0074 u0065 u0219 u0074 ], bytes: [ 74 65 c8 99 74 ]
> Received: te?t, chars: [ u0074 u0065 u003f u0074 ], bytes: [ 74 65 3f 74 ]
```

**Comment by Lolothepro** [comment]
Certain specific characters in the Windows username break compilation of project (#29213)
When the Windows username contains the "`ï`" character, it is not possible to compile NeoForge mods.

**Comment by softy-oniguru** [comment]
Of course, I apologize for getting into this issue. - I also have this error
Path: C:\.gradle\*
